### PR TITLE
Make slave mobs react to master's chase target

### PR DESCRIPTION
**Pull request by skyleo**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Make slave monsters chase after what their master is chasing, instead of merely following him until he attacks or uses a skill on a target.

That way if the master is immobilized during chasing,
the slaves will still continue to chase and attack once getting in range.

Before this fix Hunters could Ankle Snare Orc Hero for example,
making his slaves completely unresponsive for the trap duration.

**Change has been tested and found to be working well**

**Issues addressed:** None?

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2561)

File: src/map/mob.c [hunk 1]
```diff
@@ -1341,9 +1341,20 @@ static int mob_ai_sub_hard_slavemob(struct mob_data *md, int64 tick)
...
+				nullpo_retr(1, m_md);
+				switch (m_md->state.skillstate) {
+				case MSS_BERSERK:
+				case MSS_ANGRY:
+				case MSS_RUSH:
+				case MSS_FOLLOW:
+					break;
+				default:
+					return 1; // master not involved in battle, no need to pick target
+				}
```
Comment (4144) [inline src/map/mob.c]: probably better extract this code into separate function?
and also add config option for switch to use old and new way for slaves?

Comment (skyleo) [reply src/map/mob.c]: Can do so, although my sources told me its official behavior.

Comment (skyleo) [reply src/map/mob.c]: Done, check newer commits.

**Comment by 4144** [comment]
better squash fixup commit with actual commits with changes. I mean better not create separate commits with compilation fixes

**Comment by skyleo** [comment]
> 
> 
> better squash fixup commit with actual commits with changes. I mean better not create separate commits with compilation fixes

Yes, those will be squashed, but for the review they're there atm, so people aren't confused by me rebasing and changing the history.
There will be only 2 commits at the end of this Pull Request.

**Comment by skyleo** [comment]
We're assuming safely at a few places that the master is of type BL_MOB in mob->ai_sub_hard_slavemob in those changes. If you know of cases where this is not the case, please tell me so I can modify the changes.

**Comment by skyleo** [comment]
Squashed all that needed to be squashed ~
Ready to be merged.

**Pull request merged**